### PR TITLE
Add multiplatform support

### PR DESCRIPTION
**Pull request by mverleg**
This is a simple multiplatform project that can be used as a sample, and to work on issue #58.

So currently _the linting does not work_ on common or JS platforms! It does builds and run everything, and does linting for JVM.

I think this can either be left open until #58 is resolved to serve as example, or be merged now to help with testing #58.

### CODE REVIEW

**Comment by Tapchicoma** [comment]
Or you can enable _Allow edits from maintainers_, then I can try to update this PR.

**Comment by mverleg** [comment]
I think that setting is on. It's going to be a few days to look at it myself, so very welcome to make changes.

**Comment by Tapchicoma** [comment]
- Moved multiplatform samples to use the same root samples project.
- Convert them to use Kotlin DSL in build scripts.
- Add support for multiplatform kotlin plugin in plugin itself.

**Comment by Tapchicoma** [comment]
@JLLeitschuh can you check this PR? Thanks.

File: samples/kotlin-multiplatform-js/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import org.jlleitschuh.gradle.ktlint.KtlintExtension
+
+apply {
+    plugins.apply("org.jlleitschuh.gradle.ktlint")
```
Comment (JLLeitschuh) [inline samples/kotlin-multiplatform-js/build.gradle.kts]: Uhhh... This doesn't seem right.

File: samples/kotlin-multiplatform-jvm/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import org.jlleitschuh.gradle.ktlint.KtlintExtension
+
+apply {
+    plugins.apply("org.jlleitschuh.gradle.ktlint")
```
Comment (JLLeitschuh) [inline samples/kotlin-multiplatform-jvm/build.gradle.kts]: This also doesn't seem right.

File: samples/kotlin-multiplatform-js/build.gradle.kts (hunk 1 above)
Comment (Tapchicoma) [reply samples/kotlin-multiplatform-js/build.gradle.kts]: This is how all samples are implemented, check this line in main `build.gradle.kts`: https://github.com/JLLeitschuh/ktlint-gradle/blob/27b5e6586c1b1973ba19209b913ea53a3ff583cd/build.gradle.kts#L8

Comment (Tapchicoma) [reply samples/kotlin-multiplatform-js/build.gradle.kts]: So each sample module apply plugin from included plugin project using `plugins.apply()`

Comment (JLLeitschuh) [reply samples/kotlin-multiplatform-js/build.gradle.kts]: I think that this is what you want instead.
```kotlin
plugins {
    // This is a better API as the Kotlin DSL can automatically generate accessors for the `KtlintExtension` 
    id("org.jlleitschuh.gradle.ktlint")
}
apply {
    plugin("kotlin-platform-js")
}
```

If you use the above syntax, you should get this for free:
```kotlin
ktlint {
    // Your configuration goes here.
}
```

Let me know if this isn't the case.

Comment (Tapchicoma) [reply samples/kotlin-multiplatform-js/build.gradle.kts]: But will this:

Comment (Tapchicoma) [reply samples/kotlin-multiplatform-js/build.gradle.kts]: Sorry, maybe I was not clear. 

Current approach uses compiled plugin from plugin project included build: https://github.com/JLLeitschuh/ktlint-gradle/blob/27b5e6586c1b1973ba19209b913ea53a3ff583cd/settings.gradle.kts#L40

So when someone changes something in plugin project samples will compile it and apply to their configuration.

By using `plugins {}` block samples will always try to download latest plugin version from some artifactory and I can't find a way to tell it to use included build.

Also I think this part of Gradle documentation relates to this particular use-case: https://docs.gradle.org/4.5.1/userguide/plugins.html#sec:applying_plugins_buildscript
And it advises to use current approach.

Comment (JLLeitschuh) [reply samples/kotlin-multiplatform-js/build.gradle.kts]: Ah, I see. Okay.
This is related to this issue:
https://github.com/gradle/gradle/issues/2528

Okay, fair enough.
Regardless, `plugins` doesn't need to be inside of the `apply` as the receiver of the `plugins` call is on `Project`.
So you can do this instead:
```kotlin
apply {
    plugin("kotlin-platform-js")
}
plugins.apply("org.jlleitschuh.gradle.ktlint")
```

**Comment by Tapchicoma** [comment]
@JLLeitschuh addressed review

**Comment by JLLeitschuh** [comment]
Awesome! Merge at will!

**Pull request merged**